plugins/module_utils/module_helper.py

```diff
@@ -119,8 +121,11 @@ def wrapper(self, *args, **kwargs):
         except SystemExit:
             raise
         except ModuleHelperException as e:
+            self.vars.msg = e.msg
+            self.vars.exception = traceback.format_exc()
```

Comment:
What's the reasoning of putting these also into `self.vars`, and returning them both as part of `self.vars` and directly?

Reply:
Well, this part of handling variables and result values is not good yet, I intend to rewrite at some point (sooner rather than later). In the meantime, I am undoing these lines above and adjusting the lines below.